### DISCUSSION TITLE
Suggestion for new item: make:log

### DISCUSSION THREAD

**Discussion by parsampsh** [laravel/framework#38287] [Ideas]
Like requests, policies... it would be a good idea to add a new item for logs.
To refactor ours logs to several classes and then calling them.
We can do it by creating the classes in `App\Logs` namespace for example.
But it will be better if you add a command named `make:log`. Is it?

**Comment by parsampsh** [comment]
Something like:

```php
namespace App\Logs;

// use ...

class MyLog extends BaseLog {
    protected $message = 'message of the log';
    protected $level = 'info';

    public function getContext($param1, $param2...) {
        return [
            'foo' => $param1,
            'bar' => $param2,
            // ...
        ];
    }
}

////////////

\App\Logs\MyLog::log($param1, $param2...);
```

Other solution:

```php
namespace App\Logs;

// use ...

class PostLog {
    public function showed($param1, $param2)
    {
        $this->info('message', [...]);
    }

    public function deleted($param1, $param2)
    {
        $this->info('message', [...]);
    }
}

```

**Reply by dennisprudlo** [reply]
I think custom log classes with logic is something that could be a package but I wouldn't add it to the core.

I'm not sure how many Laravel developers would use this kind of approach to log.

**Reply by parsampsh** [reply]
> I think custom log classes with logic is something that could be a package but I wouldn't add it to the core.
> 
> I'm not sure how many Laravel developers would use this kind of approach to log.

So its only an option for developers. Like policies. Only for facilitating developing process.

**Reply by dennisprudlo** [reply]
Yes but what I mean is that – in my humble opinion – this doesn't need to be in the core.

A lot of developers use service classes to structure the code but that doesn't mean Laravel has to support this out of the box with a predefined folder structure and a `make:service` command.

**Reply by parsampsh** [reply]
> Yes but what I mean is that – in my humble opinion – this doesn't need to be in the core.
> 
> A lot of developers use service classes to structure the code but that doesn't mean Laravel has to support this out of the box with a predefined folder structure and a `make:service` command.

Makes sense, but I still think its a good idea :)

**Comment by shaedrich** [comment]
Would't `make:logger` be better since with `make:log` I would either expect this command to add a file to `storage/logs` or an entry to `config/logging.php`?

Even though, there might be projects where this requirement comes up, I'd say that's a very narrow use case which can usually worked around by [configuration](https://laravel.com/docs/logging#configuration).
